proto/poktroll/application/application.proto

```diff
@@ -15,4 +15,23 @@ message Application {
   cosmos.base.v1beta1.Coin stake = 2; // The total amount of uPOKT the application has staked
   repeated poktroll.shared.ApplicationServiceConfig service_configs = 3; // The list of services this appliccation is configured to request service for
   repeated string delegatee_gateway_addresses = 4 [(cosmos_proto.scalar) = "cosmos.AddressString", (gogoproto.nullable) = false]; // The Bech32 encoded addresses for all delegatee Gateways, in a non-nullable slice
+  repeated ArchivedDelegations archived_delegations = 5 [(gogoproto.nullable) = false];
 }
+
+// ArchivedDelegations defines the type used to store the archived delegations
+// for an application in addition to the last block height where the delegation
+// was active.
+message ArchivedDelegations {
+  int64 last_active_block_height = 1;
+  repeated string gateway_addresses = 2 [(cosmos_proto.scalar) = "cosmos.AddressString", (gogoproto.nullable) = false];
+}
+
+// ApplicationsWithArchivedDelegations defines the type used to store the addresses
+// of applications that have delegations being archived at lastActiveBlockHeight.
+// It is used to retrieve the applications that need to have their archived
+// delegations pruned from their corresponding application and avoid the need
+// to iterate over all applications to find the ones concerned.
+message ApplicationsWithArchivedDelegations {
```

Comment:
Does this need to be a protobuf? It seems like a structure that helps with business logic (rather that on-chain storage), so I think a struct is what we need, not a protobuf.